### ISSUE TITLE
Stabilize the single-shift logit transform for large positive and negative theta

### ISSUE THREAD

**Issue by haasonsaas** [extropic-ai/torx#16]
## Severity

**Medium–high** — `param_shift_single` overflows at ordinary saturated logits.

## Affected version

- PyPI `extro-torx==0.0.1`
- Commit `769d2f90abdfda14798fceb521143f4b99d370da`

## Summary

The shifted logit is computed as:

```python
# torx/psc/simulation/sampled.py
-jnp.log((1 + jnp.exp(-theta)) ** 2 - 1)
```

This suffers both overflow and cancellation.

## Verified (float32)

| θ | unstable | stable `θ - logaddexp(log 2, -θ)` |
|---|----------|-------------------------------------|
| −100 | `-inf` (should be −200) | −200 |
| 15 | 14.1506 (should be ≈14.3069) | correct |
| 17 | `+inf` (should be ≈16.3069) | correct |
| 20 | `+inf` (should be ≈19.3069) | correct |

Existing gradient tests only cover θ ∈ {−2, 0, 2}.

## Suggested fix

```python
log2 = jnp.log(jnp.asarray(2, dtype=theta.dtype))
shift = theta - jnp.logaddexp(log2, -theta)
```

## Regression tests

Finite-gradient / finite-shift checks for θ across roughly `[-100, 100]`.
